docs/csharp/roslyn-sdk/source-generators-overview.md

```diff
@@ -0,0 +1,158 @@
+---
+title: Source Generators
+description: Source Generators is a C# compiler feature that lets C# developers inspect user code as it is being compiled and generates new C# source files on the fly that are added to the user's compilation.
```

Comment:
Do you want to add more keywords here? Compiler Plaform, Roslyn, etc. These are prime areas for the search engine to key off of.